### PR TITLE
doc(whl):trex docs

### PR DESCRIPTION
**Pull request by kxzxvbk**
trex docs

### CODE REVIEW

File: source/hands_on/trex.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+|                     |  reward 200     |                                                     |lunarlander_trex_dqn_     |
+|                     |                 |                                                     |config.py>`_              |
++---------------------+-----------------+-----------------------------------------------------+--------------------------+
+|                     |                 |                                                     |`config_link_h <https://  |
+|                     |                 |                                                     |github.com/opendilab/     |
+|Hopper               |                 |                                                     |DI-engine/tree/main/dizoo/|
+|                     |  3M  env_step,  |.. image:: images/benchmark/Hopper_gcl.png           |mujoco/config/            |
+|                     |  reward 2950    |                                                     |.py>`_                    |
++---------------------+-----------------+-----------------------------------------------------+--------------------------+
```
Comment (Weiyuhong-1998) [inline source/hands_on/trex.rst:76]: 这儿不是放GCL的实验图，是说等以后你做了Trex的实验后放Trex的实验图

File: source/hands_on/trex.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+3. Usually, multiple data augmentation methods should be applied to prevent reward network from over-fitting.
+
+Key Equations or Key Graphs
+---------------------------
+The loss for reward function:
+
+.. math::
+
+   \mathcal{L}(\theta)=\mathbb{E}_{\tau_i, \tau_j \sim \Pi}[\xi(\mathcal{P}(\hat J_\theta(\tau_i) < \hat J_\theta(\tau_j)),\tau_i \prec \tau_j)]
```
Comment (Will-Nie) [inline source/hands_on/trex.rst:22]: It might be better to explain more about how these ranking traj are generated, especially in out DI-engine implementation (i.e. the experiments you carried out below)

File: source/hands_on/trex.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    - PPO `Proximal Policy Optimization <https://arxiv.org/pdf/1707.06347.pdf>`_
+
+    - SAC `Soft Actor-Critic <https://arxiv.org/pdf/1801.01290>`_
+
+    Given demonstrations generated from RL algorithms or human knowledge (only **observations** and **rankings** are needed), TREX will infer the reward function of the environment. Then the reward function can be applied to RL algorithms like PPO or SAC to estimate rewards while training.
+
+Implementations
+----------------
+The input of the reward model is observations and its output is the predicted reward value. The default reward model is defined as follows:
```
Comment (Will-Nie) [inline source/hands_on/trex.rst:53]: It might be better to put more implementation details about how we create the training data set

File: source/hands_on/trex.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
+TREX
+^^^^^^^
+
+Overview
+---------
+TREX (Trajectory-ranked Reward Extrapolation) was first proposed in `Extrapolating Beyond Suboptimal Demonstrations via Inverse Reinforcement Learning from Observations <https://arxiv.org/abs/1904.06387>`_, which uses ranked demonstrations to inference reward function. Different from previous methods, TREX seeks a reward function that explains the ranking over demonstrations, rather than justifying the demonstrations. Hence, this algorithm can learn good policies from highly suboptimal demostrations.
+
+Quick Facts
```
Comment (Will-Nie) [inline source/hands_on/trex.rst:8]: On key idea, which also indicated on the title of the paper, is the extrapolation beyond the suboptimal demonstrations data. Would you please also emphasise this key point here?

Comment (Will-Nie) [inline source/hands_on/trex.rst:8]: Another key point is also indicated from the title that this reward function only depends on observations, not actions. I guess it is worth adding here

File: source/hands_on/trex.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+| environment         |best mean reward | evaluation results                                  | config link              |
++=====================+=================+=====================================================+==========================+
+|                     |                 |                                                     |`config_link_l <https://  |
+|                     |                 |                                                     |github.com/opendilab/     |
+|                     |                 |                                                     |DI-engine/tree/main/dizoo/|
+|Lunarlander          |  2M env_step,   |.. image:: images/benchmark/lunarlander_gcl.png      |box2d/lunarlander/config/ |
+|                     |  reward 200     |                                                     |lunarlander_trex_dqn_     |
+|                     |                 |                                                     |config.py>`_              |
++---------------------+-----------------+-----------------------------------------------------+--------------------------+
```
Comment (Will-Nie) [inline source/hands_on/trex.rst:70]: adding something that indicates the features of Trex algo. For instance, what is the best suboptimal data you used?

File: source/hands_on/trex.rst (hunk 4 above)
Comment (Will-Nie) [inline source/hands_on/trex.rst:8]: Would you also comment on the final part of the paper, i.e. noisy ranking, ranking along with time

File: source/hands_on/trex.rst [hunk 6]
```diff
@@ -0,0 +1,90 @@
+TREX
+^^^^^^^
+
+Overview
+---------
+TREX (Trajectory-ranked Reward Extrapolation) was first proposed in `Extrapolating Beyond Suboptimal Demonstrations via Inverse Reinforcement Learning from Observations <https://arxiv.org/abs/1904.06387>`_, which uses ranked demonstrations to inference reward function. Different from previous methods, TREX seeks a reward function that explains the ranking over demonstrations, rather than justifying the demonstrations. Hence, this algorithm can learn good policies from highly suboptimal demostrations. In addition, the predicted reward only depends on observations, rather than actions.
+
```
Comment (Will-Nie) [inline source/hands_on/trex.rst:7]: 'Hence, this algorithm can learn good policies from highly suboptimal demostrations'. Why does ranking lead to this feature? this 'Hence' might not be very obvious to audiences.

Also, there is a typo in 'demonstrations'

File: source/hands_on/trex.rst [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+Quick Facts
+-------------
+1. TREX is an Inverse Reinforcement Learning (IRL) algorithm and can be combined with any RL algorithm.
+
+2. Demonstrations used in TREX require **ranking** information.
+
+3. The reward function optimization can be viewed as a simple binary classification problem.
+
+4. TREX is able to infer a meaningful reward function even when noisy, time-based rankings are provided.
```
Comment (Will-Nie) [inline source/hands_on/trex.rst:16]: might be better to add Trex can be combined with any rl algorithm
